### PR TITLE
Fix Signing Substrate Payload, Better UX for Errors

### PR DESCRIPTION
**Pull request by yjkimjunior**
#339

also added the ios production build script to package.json

### CODE REVIEW

File: src/util/db.js [hunk 1]
```diff
@@ -54,12 +54,14 @@ function txKey(hash) {
 export const deleteAccount = async account =>
   SecureStorage.deleteItem(accountId(account), accountsStore);
 
-export const saveAccount = account =>
+export const saveAccount = account => {
+  console.log('account -> ', account);
```
Comment (yjkimjunior) [inline src/util/db.js]:
```suggestion
```

**Comment by yjkimjunior** [comment]
Hmm I will take a look

File: src/util/account.js [hunk 2]
```diff
@@ -4,14 +4,16 @@ export function accountId({
...
-  const { ethereumChainId, protocol, genesisHash } = NETWORK_LIST[networkKey];
-
-  if (typeof address !== 'string' || address.length === 0 || !networkKey) {
-    throw new Error(`Couldn't create an accountId, address or networkKey missing`);
+  if (typeof address !== 'string' || address.length === 0 || !networkKey || !NETWORK_LIST[networkKey]) {
+    throw new Error(`Couldn't create an accountId. Address or networkKey missing, or network key was invalid.`);
   }
 
-  if (protocol === NetworkProtocols.SUBSTRATE){ 
+  const { ethereumChainId='', protocol, genesisHash } = NETWORK_LIST[networkKey];
```
Comment (hanwencheng) [inline src/util/account.js]: Here I guess `genesisHash` may also be undefined for Ethereum transactions.

Comment (Tbaut) [reply src/util/account.js]: indeed, since it's not used anyway it doesn't crash, but it should probably be initialized as well.

Comment (Tbaut) [reply src/util/account.js]: addressed in https://github.com/paritytech/parity-signer/pull/346

**Pull request merged**